pkg/freeze/crio/pause.go

```diff
@@ -0,0 +1,85 @@
+package crio
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"knative.dev/container-freezer/pkg/freeze/common"
+)
+
+const defaultCrioAddress = "/var/run/crio/crio.sock"
+
+func NewCrioProvider() (*CrioCRI, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, defaultCrioAddress, grpc.WithInsecure(), grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(1024*1024*16)), grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+		return (&net.Dialer{}).DialContext(ctx, "unix", addr)
+	}))
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", defaultCrioAddress)
+			},
+		},
+	}
+
+	return &CrioCRI{conn: conn, crio: client}, nil
+}
+
+type CrioCRI struct {
+	conn *grpc.ClientConn
+	crio *http.Client
```

Comment:
Would it make sense to define a `type crioClient http.Client` to use here?

Reply:
U mean:
```
type CrioCRI struct {
	conn *grpc.ClientConn
	crioClient *http.Client
}
```
Like this?

Reply:
I was thinking along these lines:

```golang
type crioClient http.Client

type CrioCRI struct {
    conn *grpc.ClientConn
    crio *crioClient
}
```

(might be overkill though... wdyt?)

Reply:
Yeah, it's better idea, but if change like this, Pause function will do type convert:
```go
func (c *CrioCRI) Pause(ctx context.Context, container string) error {
+	client :=  c.crioClient.(*http.Client)
+	resp, err :=client.Get("http://localhost/pause/" + container)
-       resp, err := c.crioClient.Get("http://localhost/pause/" + container)
	if err != nil {
		return fmt.Errorf("%s not paused: %v", container, err)
	}

	if resp.StatusCode != http.StatusOK {
		errInfo, err := ioutil.ReadAll(resp.Body)
		if err != nil {
			return fmt.Errorf("%s not paused:%v", container, err)
		}
		return fmt.Errorf("%s not paused:%v", container, string(errInfo))
	}

	return nil
}
```
looks it's better to keep like this, or maybe there is a  better way?

Reply:
Ok, with the type conversion agree it's probably best to leave it as is.